src/framework/audio/internal/soundtracks/soundtrackwriter.cpp

```diff
@@ -160,6 +161,11 @@ Ret SoundTrackWriter::generateAudioData()
 
         inputBufferOffset += samplesToCopy;
         sendStepProgress(PREPARE_STEP, inputBufferOffset, inputBufferMaxOffset);
+
+        //! NOTE It is necessary for cancellation to work
+        //! and for information about the audio signal to be transmitted.
+        async::processEvents();
```

Comment:
Do you know why this is necessary? Can we only call rpcChannel()->process()?

Reply:
The problem is that the mixer mixes not in one worker thread, but in several threads.
In order for audio signals from these threads to reach through async channels, we need to call async::processEvents(), then the audio signal (or any other data) will reach and be sent through the RPC to main